app/Foliage/PreparePackageVersion.hs

```diff
@@ -111,22 +187,29 @@ preparePackageVersion inputDir metaFile = do
           "version in cabal file: " ++ prettyShow (Distribution.Types.PackageId.pkgVersion $ package $ packageDescription pkgDesc)
         ]
 
-  let cabalFileRevisions =
-        sortOn
-          (Down . fst)
-          [ (revisionTimestamp, cabalFileRevisionPath revisionNumber)
-            | RevisionSpec {revisionTimestamp, revisionNumber} <- packageVersionRevisions pkgSpec
-          ]
+  let cabalFileRevisions = sortOn Down [(revisionTimestamp, cabalFileRevisionPath revisionNumber) | RevisionSpec {revisionTimestamp, revisionNumber} <- packageVersionRevisions pkgSpec]
+
+  let pkgVersionDeprecationChanges = sortOn Down [(deprecationTimestamp, deprecationIsDeprecated) | DeprecationSpec {deprecationTimestamp, deprecationIsDeprecated} <- packageVersionDeprecations pkgSpec]
```

Comment:
It's a little funny that these are sorted in reverse chronological order here, but later when you make the entries you re-sort them in chronological order. Seems sensible, though, you make sure they're in the order you want when you consume them.

Reply:
WUT? I am not sure why we are sorting them in different ways. I might need to double check.

Reply:
It seems sensible to me: here you look at them to see the "latest" state, when constructing the index you instead process them in chronological order.